### PR TITLE
More edge property filters

### PR DESCRIPTION
**Pull request by ljeub-pometry**
### What changes were proposed in this pull request?

- filter based on property existence
- filter based on set of values
- `ne`, `ge`, `le`

### Why are the changes needed?

we need to filter properties

### Does this PR introduce any user-facing change? If yes is this documented?

yes, more methods with python doc strings

### How was this patch tested?

needs more tests

### Are there any further changes required?

more tests

### CODE REVIEW

File: raphtory/src/core/entities/properties/props.rs [hunk 1]
```diff
@@ -275,6 +275,28 @@ impl PropMapper {
...
     }
+
+    pub fn get_and_validate(
+        &self,
+        prop: &str,
+        dtype: PropType,
+    ) -> Result<Option<usize>, GraphError> {
+        match self.get_id(prop) {
+            Some(id) => {
+                let existing_dtype = self.get_dtype(id).unwrap();
```
Comment (miratepuffin) [inline raphtory/src/core/entities/properties/props.rs:286]: I know this cannot fail, but worth turning the unwrap into a '?'

File: raphtory/src/db/api/view/edge_property_filter.rs [hunk 2]
```diff
@@ -1,78 +1,207 @@
...
+        value: Prop,
+    ) -> Result<Self::FilteredViewType, GraphError> {
+        let (t_prop_id, c_prop_id) =
+            get_ids_and_check_type(self.current_filter().edge_meta(), property, value.dtype())?;
+
+        Ok(self.one_hop_filtered(EdgePropertyFilteredGraph::new(
+            self.current_filter().clone(),
+            t_prop_id,
+            c_prop_id,
+            PropValueFilter::new(value, |left, right| left < right),
```
Comment (miratepuffin) [inline raphtory/src/db/api/view/edge_property_filter.rs:148]: Is there a reason these all read right to left?

**Review by miratepuffin — APPROVED** [review]
Couple minor points/questions but otherwise LGTM

File: raphtory/src/core/entities/properties/props.rs (hunk 1 above)
Comment (ljeub-pometry) [reply raphtory/src/core/entities/properties/props.rs:286]: Can turn it into an expect, if this is ever not true, we want shit to blow up!

File: raphtory/src/db/api/view/edge_property_filter.rs (hunk 2 above)
Comment (ljeub-pometry) [reply raphtory/src/db/api/view/edge_property_filter.rs]: Swapped the arguments around so this is less confusing

File: raphtory/src/db/api/view/edge.rs [hunk 3]
```diff
@@ -145,16 +145,11 @@ impl<'graph, E: BaseEdgeViewOps<'graph>> EdgeViewOps<'graph> for E {
 
     /// list the activation timestamps for the edge
     fn history(&self) -> Self::ValueType<Vec<i64>> {
-        self.map(|g, e| g.edge_history(e, g.layer_ids().constrain_from_edge(e)))
+        self.map(|g, e| g.edge_history(e, g.layer_ids()).map(|ti| ti.t()).collect())
```
Comment (fabianmurariu) [inline raphtory/src/db/api/view/edge.rs:148]: No iter here?

File: raphtory/src/db/api/view/edge_property_filter.rs
Comment (fabianmurariu) [inline raphtory/src/db/api/view/edge_property_filter.rs:1]: Much repetition, this file has

**Review by fabianmurariu — APPROVED** [review]
LGTM

**Pull request merged**